plugin/trino-hive/src/test/java/io/trino/plugin/hive/AbstractTestHiveFileSystem.java

```diff
@@ -400,6 +406,65 @@ public void testRename()
         fs.delete(basePath, true);
     }
 
+    @Test
+    public void testFileIteratorListing()
+            throws Exception
+    {
+        Table.Builder tableBuilder = Table.builder()
+                .setDatabaseName(table.getSchemaName())
+                .setTableName(table.getTableName())
+                .setDataColumns(ImmutableList.of(new Column("one", HIVE_LONG, Optional.empty())))
+                .setPartitionColumns(ImmutableList.of())
+                .setOwner(Optional.empty())
+                .setTableType("fake");
+        tableBuilder.getStorageBuilder()
+                .setStorageFormat(StorageFormat.fromHiveStorageFormat(HiveStorageFormat.CSV));
+        Table fakeTable = tableBuilder.build();
+
+        Path basePath = new Path(getBasePath(), "test-file-iterator-listing");
+        FileSystem fs = hdfsEnvironment.getFileSystem(TESTING_CONTEXT, basePath);
+        fs.mkdirs(basePath);
+
+        // create file in hidden folder
+        Path hiddenFile = new Path(new Path(basePath, ".hidden"), "hidden-file.txt");
```

Comment:
You may want to add different combinations of hidden files:
- in hidden directory
- in hidden subdirectory

in order to thoroughly test `io.trino.plugin.hive.fs.HiveFileIterator#isHiddenOrWithinHiddenParentDirectory`

Reply:
I've added some extra hidden files scenarios to the test, although in general these are easier to exercise via `TestHiveFileIterator`. This new test is basically just attempting to make sure that the various supported `FileSystem` implementations actually respect the `listFiles(Path, boolean)` contract.